ibm/service/iamidentity/resource_ibm_iam_account_settings_template.go

```diff
@@ -0,0 +1,658 @@
+// Copyright IBM Corp. 2023 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package iamidentity
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/iamidentityv1"
+)
+
+func ResourceIBMAccountSettingsTemplate() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMAccountSettingsTemplateCreate,
+		ReadContext:   resourceIBMAccountSettingsTemplateRead,
+		UpdateContext: resourceIBMAccountSettingsTemplateUpdate,
+		DeleteContext: resourceIBMAccountSettingsTemplateDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the account where the template resides.",
+			},
```

Comment:
All the arguments in a resource are optional 
?

resource "ibm_iam_account_settings_template" "tmp" {
}
will work?

Reply:
Can we add some parameters like atleastOneof ExactlyOneof based on minimun arguments

Reply:
Done.